addons/mail/sfu_server/src/models/channel.js

```diff
@@ -0,0 +1,224 @@
+        if (oldChannel) {
+            logger.info(`reusing channel ${oldChannel.uuid}`);
+            return oldChannel;
+        }
+        const { worker, releaseWorker } = getWorker();
+        const router = await worker.createRouter({
+            mediaCodecs,
+        });
+        const channel = new Channel(remoteAddress, { worker, router });
+        Channel.recordsByIssuer.set(safeIssuer, channel);
+        logger.info(`created channel ${channel.uuid} for ${remoteAddress}`);
+        channel.once("close", () => {
+            releaseWorker();
+            Channel.recordsByIssuer.delete(safeIssuer);
+        });
+        channel.setCloseTimeout(true);
+        return channel;
+    }
+
+    /**
+     * @param {string} uuid
+     * @param {number} sessionId
+     */
+    static join(uuid, sessionId) {
+        const channel = Channel.records.get(uuid);
+        if (!channel) {
+            throw new Error(`channel [${uuid}] does not exist`);
+        }
+        const session = channel.join(sessionId);
+        return { channel, session };
+    }
+
+    static closeAll() {
+        for (const channel of Channel.records.values()) {
+            channel.close();
+        }
+        Channel.records.clear();
+    }
+
+    /**
+     * @returns {{ channelCount: number, sessionCount: number}}
+     */
+    static getStats() {
+        return {
+            channelCount: Channel.records.size,
+            sessionCount: [...Channel.records.values()].reduce(
+                (acc, channel) => acc + channel.sessions.size,
+                0
+            ),
+        };
+    }
+
+    /**
+     * @param {string} remoteAddress
+     * @param {Object} param1
+     * @param {import("mediasoup").types.Worker} param1.worker
+     * @param {import("mediasoup").types.Router} param1.router
+     */
+    constructor(remoteAddress, { worker, router }) {
+        super();
+        this.remoteAddress = remoteAddress;
+        this.uuid = crypto.randomUUID();
+        this.name = `${remoteAddress}*${this.uuid.slice(-5)}`;
+        this.router = router;
+        this._worker = worker;
+        this._onSessionClose = this._onSessionClose.bind(this);
+        Channel.records.set(this.uuid, this);
+    }
+
+    /**
+     * @return {Promise<{total: number, screen: number, audio: number, camera: number}>}
+     */
+    async getIncomingBitrate() {
+        let audioSum = 0;
+        let cameraSum = 0;
+        let screenSum = 0;
+        for (const session of this.sessions.values()) {
+            const { audio, camera, screen } = await session.getProducerBitRates();
+            audioSum += audio || 0;
+            cameraSum += camera || 0;
+            screenSum += screen || 0;
+        }
+        return {
+            audio: audioSum,
+            camera: cameraSum,
+            screen: screenSum,
+            total: audioSum + cameraSum + screenSum,
+        };
+    }
+
+    /**
+     * @returns {Object<number, Object>}
```

Comment:
Object is not a generic type in typescript, you want `Record`. The inner object type should be more precise. The key type will be coerced to strings in objects in any case so number doesn't work